### ISSUE TITLE
Different configurations for different file extensions

### ISSUE THREAD

**Issue by enessoylu** [Microsoft/vscode-tslint#420]
I have different rule set for test files and code files. For example, I am forbidding the magic constants for codes but not for the test files. Also I have different naming conventions among the TSX files and TS files. So it would be great to have an option that maps glob patterns to configuration files. If there is no match with that patters the default config file wich is set with 'tslint.configFile' option may be used.

**Comment by egamma** [comment]
@enessoylu I think this functionality should be provided at the level of tslint and not at the level of vscode-tslint.

**Comment by enessoylu** [comment]
@egamma TSLint has the option for input files. I can pass different configurations for different files. But in VSCode I can only use one configuration for the entire workspace. That is something we need to change IMO.

**Comment by egamma** [comment]
@enessoylu can you point me to the option for input files you refer to above?

**Comment by enessoylu** [comment]
Just by passing the function to TSLint, what I mean is something like:
```ts
// tslintRunner.ts
if(extension(file).match('spec.ts')) {
   run('tslint --config ConfigFile' + file)
}
else if (extension(file).match('ts')) {
    run('tslint --config AnotherConfigFile' + file)
}
```
